docs/circle.rst

```diff
@@ -0,0 +1,112 @@
+
+    Here is the list of all the attributes of the Circle class:
+
+    .. attribute:: x
+        | :sl:`x coordinate of the center of the circle`
+        | :sg:`x -> float`
+        | :sg:`x = float`
+
+        The `x` coordinate of the center of the circle. It can be reassigned to move the circle.
+        Reassigning the `x` attribute will move the circle to the new `x` coordinate.
+        The `y` and `r` attributes will not be affected.
+
+    .. attribute:: y
+        | :sl:`y coordinate of the center of the circle`
+        | :sg:`y -> float`
+        | :sg:`y = float`
+
+        The `y` coordinate of the center of the circle. It can be reassigned to move the circle.
+        Reassigning the `y` attribute will move the circle to the new `y` coordinate.
+        The `x` and `r` attributes will not be affected.
+
+    .. attribute:: r
+        | :sl:`radius of the circle`
+        | :sg:`r -> float`
+        | :sg:`r = float`
+
+        It is not possible to set the radius to a negative value. It can be reassigned.
+        If reassigned it will only change the radius of the circle.
+        The circle will not be moved from its original position.
+
+    .. attribute:: r_sqr
+        | :sl:`radius of the circle squared`
+        | :sg:`r_sqr -> float`
+        | :sg:`r_sqr = float`
+
+        It's equivalent to `r*r`. It can be reassigned. If reassigned, the radius
+        of the circle will be changed to the square root of the new value.
+        The circle will not be moved from its original position.
+
+    .. attribute:: center
+        | :sl:`x and y coordinates of the center of the circle`
+        | :sg:`center -> (x, y)`
+        | :sg:`center = (x, y)`
+
+        (Virtual attribute)
+        It's a tuple containing the `x` and `y` coordinates that represent the center
+        of the circle. It can be reassigned. If reassigned, the circle will be moved
+        to the new position. The radius will not be affected.
+
+    .. attribute:: diameter, d
+        | :sl:`diameter of the circle`
+        | :sg:`diameter -> float`
+        | :sg:`diameter = float`
+
+        (Virtual attribute)
+        It's calculated using the `d=2*r` formula. It can be reassigned. If reassigned
+        the radius will be changed to half the diameter.
+        The circle will not be moved from its original position.
+
+    .. attribute:: area
+        | :sl:`area of the circle`
+        | :sg:`area -> float`
+        | :sg:`area = float`
+
+        (Virtual attribute)
+        It's calculated using the `area=pi*r*r` formula. It can be reassigned.
+        If reassigned the circle radius will be changed to produce a circle with matching
+        area. The circle will not be moved from its original position.
+
+    .. attribute:: circumference
+        | :sl:`circumference of the circle`
+        | :sg:`circumference -> float`
+        | :sg:`circumference = float`
+
+        (Virtual attribute)
```

Comment:
I think we can get rid of all `(Virtual attribute)`s

Reply:
Yeah I agree